### PR TITLE
First part of media translation (modules "photo" and "gallery")

### PR DESCRIPTION
**Pull request by strategio**
The code is based on reusable elements coming from "Page Builders" package (see other PR https://github.com/OnTheGoSystems/wpml-page-builders/pull/24).

For now, I added support to "photo" and "gallery" but I want to make sure we agree on the strategy before to go further.

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlpb-160

### CODE REVIEW

**Comment by strategio** [comment]
@brucepearson I closed the previous MRs to make it clearer, could you have a look at this and https://github.com/OnTheGoSystems/wpml-page-builders/pull/24

**Review by brucepearson — APPROVED** [review]
@strategio This looks ok but I don't really like the use of `helper` in variable names for the same reason as I mentioned in the other pull request.

File: tests/phpunit/tests/media/test-wpml-beaver-builder-media-node-provider.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	 * @param string $class_name
+	 */
+	public function it_should_return_a_node_instance_and_cache_it( $type, $class_name ) {
+		$GLOBALS['sitepress'] = $this->getMockBuilder( 'SitePress' )->disableOriginalConstructor()->getMock();
+		$this->mock_external_classes();
+
+		$subject = new WPML_Beaver_Builder_Media_Node_Provider();
+
+		$this->assertInstanceOf( $class_name, $subject->get( $type ) );
+		$this->assertSame( $subject->get( $type ), $subject->get( $type ) );
```
Comment (adrianosferreira) [inline tests/phpunit/tests/media/test-wpml-beaver-builder-media-node-provider.php:22]: @strategio what is the chance of this fail?

**Review by adrianosferreira — APPROVED** [review]
Just a question about an assertion, +1.

File: tests/phpunit/tests/media/test-wpml-beaver-builder-media-node-provider.php (hunk 1 above)
Comment (strategio) [reply tests/phpunit/tests/media/test-wpml-beaver-builder-media-node-provider.php:22]: @adrianosferreira, the idea is to store the node instances in the provider. It's not something critical but the test is supposed to reflect this.

Comment (adrianosferreira) [reply tests/phpunit/tests/media/test-wpml-beaver-builder-media-node-provider.php:22]: @strategio but my question is how can't `$subject->get( $type )` be the same as `$subject->get( $type )`?

Comment (strategio) [reply tests/phpunit/tests/media/test-wpml-beaver-builder-media-node-provider.php:22]: @adrianosferreira, look at https://github.com/OnTheGoSystems/wpml-page-builders-beaver-builder/pull/10/files#diff-183ca675c8229183325b9568713c41d5R17. For each type, we store the instance and return the same one if it's called several times. The test should fail if we don't store the instances (i.e. we would get different instances for the same type).

Comment (adrianosferreira) [reply tests/phpunit/tests/media/test-wpml-beaver-builder-media-node-provider.php:22]: @strategio oh, I didn't notice this little important thing `* @dataProvider dp_node_types` :D

**Pull request merged**